### PR TITLE
:bug: Adding tests for file searching and using relative path to base for matching files/filepaths

### PR DESCRIPTION
**Pull request by shawn-hurley**
## Summary by CodeRabbit

* **Tests**
  * Added a comprehensive table-driven test suite for file search covering patterns, wildcards, include/exclude rules, directory exclusions, rule-scope overrides, and edge cases.

* **Bug Fixes**
  * Improved search accuracy by matching paths relative to the search base and preserved abort behavior on path resolution errors so excludes remain honored.

* **Chores**
  * Enhanced verbose debug logging for pattern-based searches to aid troubleshooting.

### CODE REVIEW

File: provider/lib.go [hunk 1]
```diff
@@ -231,17 +231,28 @@ func (f *FileSearcher) filterFilesByPathsOrPatterns(statFunc cachedOsStat, patte
 					rPattern = "^" + pattern + "$"
 				}
 				// try matching as go regex pattern
+				relPath, err := filepath.Rel(f.BasePath, file)
+				if err != nil {
+					// This should never happen, if paths are found outside the base path, then something did not occur
+					// correctly.
+					return nil
```
[the remark this answers is not in the document]
Comment (shawn-hurley) [reply provider/lib.go]: yeah I'm convinced that continue is better and adding an error log message

+1
